applications/models.py

```diff
@@ -227,12 +227,16 @@ def get_by_rsvp_code(self, code, page):
     def is_accepted(self):
         return self.state == 'accepted'
 
-    def is_scored_by_user(self, user):
+    def is_scored_by_user(self, user, use_prefetched=True):
         """
         Returns true if the given user has scored this application
         or false if they have not, or there is a zero score.
         """
-        return self.scores.filter(user=user, score__gt=0).exists()
+        try:
+            assert self._prefetched_objects_cache['scores'] and use_prefetched # check if scores usable
+            return any(map(lambda s: s.user_id == user.id and s.score and s.score > 0, self.scores.all()))
+        except (AssertionError, AttributeError, KeyError):
```

Comment:
This is a lot of different error types to be catching, I don't feel like it's clear what's going on and what the code's expecting to fail. I can see that you're catching AttributeError in case `prefetched_objects_cache` isn't present, but it'll also trigger on other things - it's easy to obscure legitimate errors this way when you probably wanted to hear about them.

How about an alternative way of checking the cache, which doesn't require try/except

```python
cache = getattr(self, '_prefetched_objects_cache', {})
if 'scores' in cache:
    # etc...
```